src/chanjo2/meta/handle_d4.py

```diff
@@ -283,41 +235,63 @@ def get_report_sample_interval_coverage(
 
 def get_sample_interval_coverage(
     db: Session,
-    d4_file: D4File,
+    d4_file_path: str,
     genes: List[SQLGene],
     interval_type: Union[SQLGene, SQLTranscript, SQLExon],
     completeness_thresholds: List[Optional[int]],
     transcript_tags: Optional[List[TranscriptTag]] = [],
 ) -> List[GeneCoverage]:
```

Comment:
```suggestion
) -> List[GeneCoverage] | list[None]:
```
Since this is python 3.11, you can use lower case for buitins (and `|` instead of Optional/Union), but perhaps it would be better to use upper case to be consistent.